src/PooledArrays.jl

```diff
@@ -184,31 +211,68 @@ Base.findall(pdv::PooledVector{Bool}) = findall(convert(Vector{Bool}, pdv))
 ##
 ##############################################################################
 
-function Base.map(f, x::PooledArray{T,R}) where {T,R<:Integer}
-    ks = collect(keys(x.invpool))
-    vs = collect(values(x.invpool))
-    ks1 = map(f, ks)
-    uks = Set(ks1)
-    if length(uks) < length(ks1)
-        # this means some keys have repeated
-        newinvpool = Dict{eltype(ks1), eltype(vs)}()
-        translate = Dict{eltype(vs), eltype(vs)}()
-        i = 1
-        for (k, k1) in zip(ks, ks1)
-            if haskey(newinvpool, k1)
-                translate[x.invpool[k]] = newinvpool[k1]
+"""
```

Comment:
@nalimilan @bkamins, this is where the `droplevels!` is defined, with the new definition of `map` below it.